### DISCUSSION TITLE
issue with babel__traverse since version 7.18.3 (maybe earlier)

### DISCUSSION THREAD

**Discussion by awatt-cox** [DefinitelyTyped/DefinitelyTyped#63420] [Issues with a @types package]
I have a serverless/lambda project written in typescript that was building and deploying fine last week via a Jenkins pipeline. This week, when I run the same pipeline, it fails to build and throws a bunch of errors related to babel__traverse. Here's part of the stack trace:

Error:
Webpack compilation failed:

in /var/jenkins_home/jobs/Backend_deploy/workspace/node_modules/@types/babel__traverse/index.d.ts
  [tsl] in /var/jenkins_home/jobs/Backend_deploy/workspace/node_modules/@types/babel__traverse/index.d.ts(68,50)
        TS1005: ']' expected.
  
  in /var/jenkins_home/jobs/Backend_deploy/workspace/node_modules/@types/babel__traverse/index.d.ts
  [tsl] in /var/jenkins_home/jobs/Backend_deploy/workspace/node_modules/@types/babel__traverse/index.d.ts(68,53)
        TS1005: ';' expected.
  
  in /var/jenkins_home/jobs/Backend_deploy/workspace/node_modules/@types/babel__traverse/index.d.ts
  [tsl] in /var/jenkins_home/jobs/Backend_deploy/workspace/node_modules/@types/babel__traverse/index.d.ts(68,58)
        TS1005: ';' expected.
  
  in /var/jenkins_home/jobs/Backend_deploy/workspace/node_modules/@types/babel__traverse/index.d.ts
  [tsl] in /var/jenkins_home/jobs/Backend_deploy/workspace/node_modules/@types/babel__traverse/index.d.ts(68,70)
        TS1011: An element access expression should take an argument.
  
  in /var/jenkins_home/jobs/Backend_deploy/workspace/node_modules/@types/babel__traverse/index.d.ts
  [tsl] in /var/jenkins_home/jobs/Backend_deploy/workspace/node_modules/@types/babel__traverse/index.d.ts(68,83)
        TS1005: ';' expected.
  
  in /var/jenkins_home/jobs/Backend_deploy/workspace/node_modules/@types/babel__traverse/index.d.ts
  [tsl] in /var/jenkins_home/jobs/Backend_deploy/workspace/node_modules/@types/babel__traverse/index.d.ts(68,84)
        TS1128: Declaration or statement expected.
  
  in /var/jenkins_home/jobs/Backend_deploy/workspace/node_modules/@types/babel__traverse/index.d.ts
  [tsl] in /var/jenkins_home/jobs/Backend_deploy/workspace/node_modules/@types/babel__traverse/index.d.ts(68,88)
        TS1128: Declaration or statement expected.

in /var/jenkins_home/jobs/Backend_deploy/workspace/node_modules/@types/babel__traverse/index.d.ts
  [tsl] in /var/jenkins_home/jobs/Backend_deploy/workspace/node_modules/@types/babel__traverse/index.d.ts(68,50)
        TS1005: ']' expected.
  
  in /var/jenkins_home/jobs/Backend_deploy/workspace/node_modules/@types/babel__traverse/index.d.ts
  [tsl] in /var/jenkins_home/jobs/Backend_deploy/workspace/node_modules/@types/babel__traverse/index.d.ts(68,53)
        TS1005: ';' expected.

Since babel_traverse is a subdependency, I have very little control over the version being pulled for webpack. Are these errors related to the 7.18.3 release?

[a comment by an automated account is not shown]

**Comment by biffgaut** [comment]
I'm seeing the same behavior - opened up Discussion 63428 before I realized this was already open.

**Comment by awatt-cox** [comment]
I don't know why 7.18.3 is broken, but I was able to force a rollback to 7.18.2 by adding a resolutions section to my package.json:
"resolutions": {
    "@types/babel__traverse": "7.18.2"
  },
This isn't a long-term fix, but it got my build pipeline working again.

**Comment by awatt-cox** [comment]
I notice that activity has died on this topic. Is there any information I can provide so that someone can look into the issue? I don't understand what typescript-bot is asking for - I can provide a full stack trace of the error I get from my pipeline. 

Or will everyone with this problem just need to keep using the non-broken 7.18.2 version?
